library/network/test/susefirewall_test.rb

```diff
@@ -12,151 +12,111 @@ def reset_SuSEFirewallIsInstalled_cache
   FakeFirewall.needed_packages_installed = nil
 end
 
-# Instantiate an SF2 object
-FakeFirewall = Yast::FirewallClass.create(:sf2)
-FakeFirewall.main
+# Instantiate an Firewalld object
+FakeFirewall = Yast::SuSEFirewalldClass.new
 
 describe FakeFirewall do
 
-  describe "#SuSEFirewallIsInstalled" do
-    before(:each) do
-      reset_SuSEFirewallIsInstalled_cache
-    end
-
+  describe "#SuSEFirewallIsSelectedOrInstalled" do
     context "while in inst-sys" do
-      it "returns whether SuSEfirewall2 is selected for installation or already installed" do
+      it "returns whether SuSEfirewalld is selected for installation or already installed" do
         expect(Yast::Stage).to receive(:stage).and_return("initial").at_least(:once)
 
         # Value is not cached
         expect(Yast::Pkg).to receive(:IsSelected).and_return(true, false, false).exactly(3).times
         # Fallback: if not selected, checks whether the package is installed
-        expect(Yast::PackageSystem).to receive(:Installed).and_return(false, true).twice
+        expect(subject).to receive(:SuSEFirewallIsInstalled).and_return(false, true).twice
 
         # Selected
-        expect(subject.SuSEFirewallIsInstalled).to eq(true)
+        expect(subject.SuSEFirewallIsSelectedOrInstalled).to eq(true)
         # Not selected and not installed
-        expect(subject.SuSEFirewallIsInstalled).to eq(false)
+        expect(subject.SuSEFirewallIsSelectedOrInstalled).to eq(false)
         # Not selected, but installed
+        expect(subject.SuSEFirewallIsSelectedOrInstalled).to eq(true)
+      end
+    end
+
+    context "while on a running system or AutoYast config" do
+      it "returns whether SuSEfirewalld was or could have been installed" do
+        expect(Yast::Stage).to receive(:stage).and_return("normal").twice
```

Comment:
You are setting preconditions, right? In that case, I would use `allow`.

Reply:
I would maybe use different contexts:

```ruby
context "while on a running system or AutoYaST config" do
  before do
    allow(Yast::Stage).to receive(:stage).and_return("normal")
    allow(:SuSEFirewallInstalled).and_return(installed)
  end

  context "and SuSEfirewalld is not installed" do
    let(:installed) { false }

    it "returns false" do
      expect(subject.SuSEFirewallIsSelectedOrInstalled).to eq(false)
    end
  end

  context "and SuSEfirewalld is installed" do
    let(:installed) { true }

    it "returns true" do
      expect(subject.SuSEFirewallIsSelectedOrInstalled).to eq(true)
    end
  end
end
```

Reply:
I just adapted to work as it was working for SuSEFirewall2 but for SuSEFirewalld, and basically it was mainly expecting the number of calls to verify it was cached when installed.

Reply:
I would use a separate example for testing the cache. But all in one test does not look good to me.